### DISCUSSION TITLE
Is .Net 4 Taboo in PowerShell 7?

### DISCUSSION THREAD

**Discussion by drstonephd** [PowerShell/PowerShell#16890] [Q&A] [answered]
I'm trying to get information from an old TFS 2012 server.  The code works in PS ISE, but not in PowerShell 7.  I was thinking there might be a work around.  However, I don't know if this is a case where it's not supported.  

I use Add-Type for these found in a VS 2019 installation.  (I might have to get older dlls, but starting here.)  

```
Microsoft.TeamFoundation.Client.dll
Microsoft.TeamFoundation.Build.Client.dll
Microsoft.TeamFoundation.TestManagement.Client.dll
Microsoft.TeamFoundation.WorkItemTracking.Client.dll
Microsoft.VisualStudio.Services.Client.Interactive.dll
Microsoft.TeamFoundation.WorkItemTracking.Client.DataStoreLoader.dll
Microsoft.TeamFoundation.Build.Common.dll
 ```

The following line works in PS ISE, but not in VS Code. 

```
[Microsoft.TeamFoundation.Client.TfsTeamProjectCollectionFactory]::GetTeamProjectCollection($tfsCollectionUri)
```

The error returned using VS Code is: 

```
MethodInvocationException: Exception calling "GetTeamProjectCollection" with "1" argument(s): "Could not load type 'System.Diagnostics.Eventing.EventProvider' from assembly 'System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'."
```

It appears to be a .net version issue.  I used the following to get the assembly info.  

```
[appdomain]::currentdomain.GetAssemblies() | 
    ?{ $_.FullName -like 'System.Core,*' } |
    Select-Object -ExpandProperty CustomAttributes |
    ?{ $_.AttributeType -match 'Target|Description|FileVersion' } |
    Select AttributeType, ConstructorArguments    
```

In PS ISE:

```
AttributeType                                  ConstructorArguments
-------------                                  --------------------
System.Reflection.AssemblyDescriptionAttribute {"System.Core.dll"} 
System.Reflection.AssemblyFileVersionAttribute {"4.8.4470.0"}
```

In VS Code:

```
AttributeType                                      ConstructorArguments
-------------                                      --------------------
System.Runtime.Versioning.TargetFrameworkAttribute {".NETCoreApp,Version=v6.0"}
System.Reflection.AssemblyDescriptionAttribute     {"System.Core"}
System.Reflection.AssemblyFileVersionAttribute     {"6.0.21.52210"}
```

There is definitely a different system.core version.  I was very confused because both are "Version=4.0.0.0".  I don't know where to start.  I don't even know if I should start.

**Comment by doctordns** [comment, marked as answer]
Your work around is not likely to ever work. Why not just use Windows Powershell?

**Reply by drstonephd** [reply]
I wanted to know if that's the way of it.  Windows PowerShell for the old things that that rely upon a .net framework.  PowerShell 7 otherwise.  

I don't see the old frameworks going away anytime soon, but I suppose this is an edge case.
